sdk/src/asset_io.rs

```diff
@@ -69,7 +69,24 @@ pub trait CAIWriter {
     ) -> Result<Vec<HashObjectPositions>>;
 }
 
-pub trait AssetIO {
+pub trait AssetIO: Sync + Send {
+    // create instance of AssetIO handler.  The extension type is passed in so
+    // that format specific customizations can be used during manifest embedding
+    fn new(asset_type: &str) -> Self
+    where
```

Comment:
I'm curious about this interface. Is there a case where an `AssetIO` implementation would _not_ be `Sized`? I'm wondering if this trait bound should move up to `AssetIO` itself rather than being on `fn new`.

Reply:
If you move the sized to the trait you get this error on get_handler interface:

"for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>"